src/test/kotlin/com/metalbear/mirrord/MirrordPluginTest.kt

```diff
@@ -0,0 +1,213 @@
+            val pluginPath = Paths.get(System.getProperty("test.plugin.path"))
+            println("downloading IDE...")
+            ideaProcess = IdeLauncher.launchIde(
+                ideDownloader.downloadAndExtract(Ide.PYCHARM_COMMUNITY, tmpDir, Ide.BuildType.RELEASE),
+                mapOf(
+                    "robot-server.port" to 8082,
+                    "idea.trust.all.projects" to true,
+                    "robot-server.host.public" to true,
+                    "ide.show.tips.on.startup.default.value" to false
+                ),
+                emptyList(),
+                listOf(ideDownloader.downloadRobotPlugin(tmpDir), pluginPath),
+                tmpDir
+            )
+            println("waiting for IDE...")
+            waitForIgnoringError(ofMinutes(3)) { remoteRobot.callJs("true") }
+        }
+
+        @AfterAll
+        @JvmStatic
+        fun cleanUp() {
+            val xpathPage = URL("http://localhost:8082/").readText()
+            // create a file in build/reports
+            val file = File("build/reports/robot-page.html")
+            file.writeText(xpathPage)
+            ideaProcess?.destroy()
+            tmpDir.toFile().deleteRecursively()
+        }
+    }
+
+    @Test
+    @Video
+    fun testMirrordFlow() = with(remoteRobot) {
+        step("Welcome Frame") {
+            welcomeFrame {
+                steps?.openProject(System.getProperty("test.workspace"))
+            }
+        }
+        idea {
+            step("Open `app.py`") {
+                // sometimes the projectViewTree does not open, this is quite random
+                // as a workaround we try to open it again
+                var clickToOpenProject = false
+                val projectViewTree = try {
+                    projectViewTree
+                } catch (e: Exception) {
+                    leftStripe {
+                        findText("Project").click()
+                    }
+                    clickToOpenProject = true
+                    projectViewTree
+                }
+                with(projectViewTree) {
+                    if (clickToOpenProject) {
+                        waitFor(ofSeconds(30)) {
+                            hasText("test-workspace")
+                        }
+                        findText("test-workspace").doubleClick()
+                    }
+                    waitFor(ofSeconds(30)) {
+                        hasText("app.py")
+                    }
+                    findText("app.py").doubleClick()
+                    editorTabs {
+                        waitFor {
+                            isFileOpened("app.py")
+                        }
+                    }
+                }
+                step("Set up Poetry Environment") {
+                    // blue hover appears on top of the text window asking
+                    // to set up poetry environment
+                    fileIntention {
+                        val setUpPoetry = setUpPoetry
```

Comment:
Can you explain this line?

Reply:
there is a blue bar on the text window when a python file is opened in a poetry project, we click on a button in it to setup poetry quickly

Reply:
I meant more like, why do you need `val setUpPoetry`? Can't you just do `setUpPoetry.click()` directly?

Reply:
this is because it is a getter function: 
```
val setUpPoetry
        get() = find<ContainerFixture>(
            byXpath("//div[@accessiblename.key='sdk.set.up.poetry.environment']"),
            Duration.ofSeconds(20)
        )
```
i need that value again in the `waitFor`